sdk/python/feast/infra/offline_stores/mssql.py

```diff
@@ -0,0 +1,494 @@
+        }
+        for entity_name in feature_view.entities:
+            entity = registry.get_entity(entity_name, project)
+            join_keys.append(entity.join_key)
+            join_key_column = reverse_field_mapping.get(
+                entity.join_key, entity.join_key
+            )
+            entity_selections.append(f"{join_key_column} AS {entity.join_key}")
+
+        if isinstance(feature_view.ttl, timedelta):
+            ttl_seconds = int(feature_view.ttl.total_seconds())
+        else:
+            ttl_seconds = 0
+
+        assert isinstance(feature_view.input, SqlServerSource)
+
+        event_timestamp_column = feature_view.input.event_timestamp_column
+        created_timestamp_column = feature_view.input.created_timestamp_column
+
+        context = FeatureViewQueryContext(
+            name=feature_view.name,
+            ttl=ttl_seconds,
+            entities=join_keys,
+            features=features,
+            table_ref=feature_view.input.table_ref,
+            event_timestamp_column=reverse_field_mapping.get(
+                event_timestamp_column, event_timestamp_column
+            ),
+            created_timestamp_column=reverse_field_mapping.get(
+                created_timestamp_column, created_timestamp_column
+            ),
+            # TODO: Make created column optional and not hardcoded
+            query=feature_view.input.query,
+            table_subquery=feature_view.input.get_table_query_string().replace("`", ""),
+            entity_selections=entity_selections,
+        )
+        query_context.append(context)
+    return query_context
+
+
+def build_point_in_time_query(
+    feature_view_query_contexts: List[FeatureViewQueryContext],
+    min_timestamp: datetime,
+    max_timestamp: datetime,
+    left_table_query_string: str,
+    entity_df_event_timestamp_col: str,
+):
+
+    """Build point-in-time query between each feature view table and the entity dataframe"""
+    template = Environment(loader=BaseLoader()).from_string(
+        source=SINGLE_FEATURE_VIEW_POINT_IN_TIME_JOIN
+    )
+
+    # Add additional fields to dict
+    template_context = {
+        "min_timestamp": min_timestamp,
+        "max_timestamp": max_timestamp,
+        "left_table_query_string": left_table_query_string,
+        "entity_df_event_timestamp_col": entity_df_event_timestamp_col,
+        "unique_entity_keys": set(
+            [entity for fv in feature_view_query_contexts for entity in fv.entities]
+        ),
+        "featureviews": [asdict(context) for context in feature_view_query_contexts],
+    }
+
+    query = template.render(template_context)
+    return query
+
+
+# TODO: Optimizations
+#   * Use NEWID() instead of ROW_NUMBER(), or join on entity columns directly
+#   * Precompute ROW_NUMBER() so that it doesn't have to be recomputed for every query on entity_dataframe
+#   * Create temporary tables instead of keeping all tables in memory
+
+SINGLE_FEATURE_VIEW_POINT_IN_TIME_JOIN = """
```

Comment:
By the way, this is a multiple feature view PIT join. The variable had improper naming previously.